src/libData/AccountData/Account.cpp

```diff
@@ -44,6 +43,52 @@ void Account::InitStorage()
     }
 }
 
+void Account::InitContract(const vector<unsigned char>& data)
+{
+    if (data.empty())
+    {
+        m_initValJson = Json::arrayValue;
+        return;
+    }
+    Json::CharReaderBuilder builder;
+    std::unique_ptr<Json::CharReader> reader(builder.newCharReader());
+    Json::Value root;
+    string dataStr(data.begin(), data.end());
+    string errors;
+    if (reader->parse(dataStr.c_str(), dataStr.c_str() + dataStr.size(), &root,
+                      &errors))
+    {
+        m_initValJson = root;
+        for (auto v : root)
+        {
+            if (!v.isMember("vname") || !v.isMember("type")
+                || !v.isMember("value"))
+            {
+                LOG_GENERAL(
+                    WARNING,
+                    "This variable in initialization of contract is corrupted");
+                continue;
+            }
+            string vname = v["vname"].asString();
+            string type = v["type"].asString();
+
+            Json::StreamWriterBuilder writeBuilder;
+            std::unique_ptr<Json::StreamWriter> writer(
+                writeBuilder.newStreamWriter());
+            ostringstream oss;
+            writer->write(v["value"], &oss);
+            string value = oss.str();
+
+            SetStorage(vname, type, value, false);
+        }
+    }
+    else
```

Comment:
Generally speaking, we can handle these special conditions first before any expected logic in a function. For example,

```
if (data.empty()) {
    LOG_GENERAL(...);
    return;
}

if (not reader->parse()) {
   LOG_GENERAL(...);
   return;
}

// the code for InitContract
```